src/overlays/actors/ovl_En_Vali/z_en_vali.c

```diff
@@ -76,71 +82,732 @@ static ColliderCylinderInit D_80B288F0 = {
+
+        scale->y -= 0.2f * sinf((M_PI / 40) * curFrame);
+    }
+}
+
+void EnVali_PulseInsides(EnVali* this, f32 curFrame, Vec3f* scale) {
+    f32 scaleChange;
+
+    if (this->actionFunc == EnVali_Attacked) {
+        s32 scalePhase = 20 - (this->lightningTimer % 20);
+
+        if (scalePhase >= 10) {
+            scalePhase -= 10;
+        }
+
+        scale->y -= 0.13f * sinf((M_PI / 10) * scalePhase);
+    } else if (this->actionFunc == EnVali_Retaliate) {
+        scaleChange = sinf((M_PI / 10) * curFrame);
+        scale->y -= 0.18f * scaleChange;
+        scale->x -= 0.1f * scaleChange;
+        scale->z = scale->x;
+    } else if (this->actionFunc == EnVali_MoveArmsDown) {
+        scaleChange = cosf((M_PI / 50) * curFrame);
+        scale->y -= 0.18f * scaleChange;
+        scale->x -= 0.1f * scaleChange;
+        scale->z = scale->x;
+    } else if (this->actionFunc == EnVali_Stunned) {
+        scaleChange = sinf((M_PI / 10) * this->timer) * 0.08f;
+        scale->x -= scaleChange;
+        scale->y += scaleChange;
+        scale->z -= scaleChange;
+    } else {
+        if (curFrame >= 40.0f) {
+            curFrame -= 40.0f;
+        }
+
+        scale->y -= 0.13f * sinf((M_PI / 40) * curFrame);
+    }
+}
+
+s32 EnVali_SetArmLength(EnVali* this, f32 curFrame) {
+    f32 targetArmScale;
+
+    if (this->actionFunc == EnVali_FloatIdle) {
+        if (curFrame <= 10.0f) {
+            targetArmScale = curFrame * 0.05f + 1.0f;
+        } else if (curFrame > 70.0f) {
+            targetArmScale = (80.0f - curFrame) * 0.05f + 1.0f;
+        } else {
+            targetArmScale = 1.5f;
+        }
+    } else if (this->actionFunc == EnVali_Retaliate) {
+        targetArmScale = 1.0f - sinf((M_PI / 10) * curFrame) * 0.35f;
+    } else if (this->actionFunc == EnVali_MoveArmsDown) {
+        targetArmScale = 1.0f - cosf((M_PI / 50) * curFrame) * 0.35f;
+    } else if ((this->actionFunc == EnVali_Attacked) || (this->actionFunc == EnVali_Frozen)) {
+        targetArmScale = this->armScale;
+    } else {
+        targetArmScale = 1.0f;
+    }
+
+    Math_StepToF(&this->armScale, targetArmScale, 0.1f);
+
+    if (this->armScale == 1.0f) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+s32 EnVali_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx,
+                            Gfx** gfx) {
+    EnVali* this = THIS;
+    f32 curFrame;
+
+    if ((limbIndex == EN_VALI_LIMB_NUCLEUS) || (limbIndex == EN_VALI_LIMB_OUTER_HOOD) ||
+        (limbIndex == EN_VALI_LIMB_INNER_HOOD)) {
+        *dList = NULL;
+        return false;
+    } else {
+        curFrame = this->skelAnime.curFrame;
+
+        if ((limbIndex == EN_VALI_LIMB_LEFT_ARM_BASE) || (limbIndex == EN_VALI_LIMB_RIGHT_ARM_BASE)) {
+            if (EnVali_SetArmLength(this, curFrame)) {
+                Matrix_Scale(this->armScale, 1.0f, 1.0f, MTXMODE_APPLY);
+            }
+        }
+
+        return false;
```

Comment:
by change can you put the return outside the if so you only need 1?

Reply:
No, this was the only way I could get it to match.